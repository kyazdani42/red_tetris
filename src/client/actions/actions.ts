import { Action, ActionFunctionAny, createAction } from 'redux-actions';

import { AppState } from '../reducers/app';

import {
  CREATE_ROOM,
  HANDLE_KEY_PRESS,
  JOIN_ROOM,
  LEAVE_ROOM,
  SET_GAME_DATA,
  SET_KEY,
  SET_OPTIONS,
  SET_PLAYER_NAME,
  SET_ROOMS,
  SET_SOCKET,
} from '../constants';

export const setRooms: ActionFunctionAny<Action<RoomType[]>> = createAction(SET_ROOMS);

export const createRoom: ActionFunctionAny<Action<string>> = createAction(CREATE_ROOM);
export const joinRoom: ActionFunctionAny<Action<undefined>> = createAction(JOIN_ROOM);
export const leaveRoom: ActionFunctionAny<Action<undefined>> = createAction(LEAVE_ROOM);

export const setSocket: ActionFunctionAny<Action<SocketIOClient.Socket>> = createAction(SET_SOCKET);

export const setGameData: ActionFunctionAny<Action<GameProps>> = createAction(SET_GAME_DATA);
export const setKey: ActionFunctionAny<Action<keyType>> = createAction(SET_KEY);
export const handleKeyPress: ActionFunctionAny<Action<keyType>> = createAction(HANDLE_KEY_PRESS);

export const setPlayerName: ActionFunctionAny<Action<string>> = createAction(SET_PLAYER_NAME);
export const setOptions: ActionFunctionAny<Action<AppState['options']>> = createAction(SET_OPTIONS);
