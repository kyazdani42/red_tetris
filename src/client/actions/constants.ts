export const SET_ROOMS = 'SET_ROOMS';
export const JOIN_ROOM = 'HANDLE_JOIN_ROOM';

export const CREATE_ROOM = 'CREATE_ROOM';
export const LEAVE_ROOM = 'LEAVE_ROOM';

export const SET_SOCKET = 'SET_SOCKET';

export const SET_GAME_DATA = 'SET_GAME_DATA';
