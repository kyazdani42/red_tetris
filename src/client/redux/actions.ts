import { Action, ActionFunctionAny, createAction } from 'redux-actions';

import { AppState } from './reducer';

import {
  CREATE_ROOM,
  GET_SCORES,
  HANDLE_KEY_PRESS,
  JOIN_ROOM,
  LEAVE_ROOM,
  SET_ERROR,
  SET_GAME_DATA,
  SET_KEY,
  SET_MODAL,
  SET_MUSIC_PLAYING,
  SET_OPTIONS,
  SET_PLAYER_NAME,
  SET_ROOMS,
  SET_SOCKET,
  SET_TOKEN,
} from './constants';

export const setRooms: ActionFunctionAny<Action<RoomType[]>> = createAction(SET_ROOMS);

export const createRoom: ActionFunctionAny<Action<string>> = createAction(CREATE_ROOM);
export const joinRoom: ActionFunctionAny<Action<undefined>> = createAction(JOIN_ROOM);
export const leaveRoom: ActionFunctionAny<Action<undefined>> = createAction(LEAVE_ROOM);
export const setModal: ActionFunctionAny<Action<boolean>> = createAction(SET_MODAL);

export const setSocket: ActionFunctionAny<Action<SocketIOClient.Socket>> = createAction(SET_SOCKET);

export const setGameData: ActionFunctionAny<Action<GameProps>> = createAction(SET_GAME_DATA);
export const setKey: ActionFunctionAny<Action<keyType>> = createAction(SET_KEY);
export const handleKeyPress: ActionFunctionAny<Action<keyType>> = createAction(HANDLE_KEY_PRESS);

export const setPlayerName: ActionFunctionAny<Action<string>> = createAction(SET_PLAYER_NAME);
export const setError: ActionFunctionAny<Action<string | null>> = createAction(SET_ERROR);
export const setOptions: ActionFunctionAny<Action<AppState['options']>> = createAction(SET_OPTIONS);
export const setToken: ActionFunctionAny<Action<string>> = createAction(SET_TOKEN);
export const setMusicPlaying: ActionFunctionAny<Action<undefined>> = createAction(SET_MUSIC_PLAYING);

export const getScores: ActionFunctionAny<Action<undefined>> = createAction(GET_SCORES);
