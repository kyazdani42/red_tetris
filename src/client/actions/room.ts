import { Action, ActionFunctionAny, createAction } from 'redux-actions';
import { RoomType } from '../types';
import { HANDLE_JOIN_ROOM, SET_ROOMS, SET_SELECTED_ROOM } from './constants';

export const setRooms: ActionFunctionAny<Action<RoomType[]>> = createAction(SET_ROOMS);
export const setSelectedRoom: ActionFunctionAny<Action<RoomType>> = createAction(SET_SELECTED_ROOM);
export const handleJoinRoom: ActionFunctionAny<Action<undefined>> = createAction(HANDLE_JOIN_ROOM);
