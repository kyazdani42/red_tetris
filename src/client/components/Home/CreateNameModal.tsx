import * as React from 'react';
import { connect } from 'react-redux';

import { State } from '../../redux/store';
import {
  CreateRoomStyle,
  ErrorStyle,
  InputWrapper,
  LabelStyle,
  ModalStyle,
  ModalWrapper,
  NameInputStyle
} from './styles';

interface Props {
  playerName: string | null;
  handleDispatch: (username: string) => void;
  setDisplayModal: (d: boolean) => void;
}

export const CreateNameModal: React.SFC<Props> = ({ handleDispatch, setDisplayModal, playerName }) => {
  const [error, setError] = React.useState(null);
  if (playerName) {
    handleDispatch(playerName);
    return null;
  }
  return (
    <ModalWrapper onClick={handleRemoveModal(setDisplayModal)} id="modal-name">
      <ModalStyle>
        <InputWrapper>
          <LabelStyle>Pick a username</LabelStyle>
          <NameInputStyle onKeyDown={handleKeyDown(handleDispatch, setError)} autoFocus={true} />
          {error ? <ErrorStyle>{error}</ErrorStyle> : null}
        </InputWrapper>
        <CreateRoomStyle onClick={handleCreateRoom(handleDispatch, setError)}>
          Launch Game
        </CreateRoomStyle>
      </ModalStyle>
    </ModalWrapper>
  );
};

const handleRemoveModal = (setDisplayModal: any) => (e: any) => {
  if (e.target.id === 'modal-name') {
    setDisplayModal(false);
  }
};

const handleCreateRoom = (dispatch: (username: string) => void, setError: any) => (e: any) => {
  const value = e.target.previousSibling.firstChild.nextElementSibling.value;
  if (!value.length) {
    setError('please enter something');
  } else {
    dispatch(value);
  }
};

const handleKeyDown = (dispatch: (username: string) => void, setError: any) => (e: any) => {
  setError(null);
  if (e.key === 'Enter') {
    if (!e.target.value.length) {
      setError('please enter something');
    } else {
      dispatch(e.target.value);
    }
  }
};

const mapStateToProps = (state: State, ownProps: any) => ({
  playerName: state.app.playerName,
  ...ownProps
});

export default connect(mapStateToProps)(CreateNameModal);
