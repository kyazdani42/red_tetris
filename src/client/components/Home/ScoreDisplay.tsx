import * as React from 'react';
import { connect } from 'react-redux';

import { State } from '../../redux/store';

import { setScores } from '../../redux/actions';
import { ModalWrapper, ScoreRowStyle, ScoreRowWrapper, ScoresWrapper, ScoreTitle } from './styles';

interface Props {
  scores: BestScore[] | null;
  resetScores: () => void;
}

const handleClick = (resetScores: () => void) => (e: any) => {
  if (e.target.classList.contains('modal')) {
    resetScores();
  }
};

export const ScoreDisplay: React.SFC<Props> = ({ scores, resetScores }) => scores && (
  <ModalWrapper className="modal" onClick={handleClick(resetScores)}>
    <ScoresWrapper>
      <ScoreTitle>Top 25 Scores</ScoreTitle>
      <ScoreRowWrapper>
        <ScoreRowStyle>
          <span><b>Player</b></span>
          <span><b>Top Score</b></span>
          <span><b>Games Played</b></span>
          <span><b>Multiplayer Game Wins</b></span>
        </ScoreRowStyle>
        {getRows(scores)}
      </ScoreRowWrapper>
    </ScoresWrapper>
  </ModalWrapper>
) || null;

const getRows = (scores: BestScore[]) => scores.map(
  d => (
  <ScoreRowStyle>
    <span>{d.name}</span>
    <span>{d.bestScore}</span>
    <span>{d.gamesPlay}</span>
    <span>{d.multiPlayersWin}</span>
  </ScoreRowStyle>
  )
);

const mapStateToProps = (state: State) => ({
  scores: state.app.scores
});

const mapDispatchToProps = (dispatch: any) => ({
  resetScores: () => dispatch(setScores(null))
});

export default connect(mapStateToProps, mapDispatchToProps)(ScoreDisplay);
