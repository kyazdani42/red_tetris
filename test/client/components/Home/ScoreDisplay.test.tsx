// tslint:disable
import * as React from 'react';
import { shallow } from 'enzyme';

import { ScoreDisplay, mapStateToProps, mapDispatchToProps } from '../../../../src/client/components/Home/ScoreDisplay';
import { setScores } from '../../../../src/client/redux/actions';

describe('testing the component', () => {
  const props = {
    scores: [
      { bestScore: 10, name: 'testplayer', gamesPlay: 1, multiPlayersWin: 1 }
    ],
    resetScores: jest.fn()
  };
  const wrapper = shallow(<ScoreDisplay {...props} />);
  it('smoke tests the component', () => {
    expect(wrapper.update()).toMatchSnapshot();
  });
  it('simulate the click but doesnt call the function', () => {
    const mockedEvent = { target: { classList: { contains(s: string) { return this.elements.includes(s) }, elements: ['not-modal'] } } };
    wrapper.simulate('click', mockedEvent);
    expect(props.resetScores).not.toHaveBeenCalled();
  });
  it('simulate the click and calls the function', () => {
    const mockedEvent = { target: { classList: { contains(s: string) { return this.elements.includes(s) }, elements: ['modal'] } } };
    wrapper.simulate('click', mockedEvent);
    expect(props.resetScores).toHaveBeenCalled();
  });
  it('return null when there are no scores', () => {
    wrapper.setProps({ scores: null, resetScore: null });
    expect(wrapper.exists('div')).toBeFalsy();
  })
});

describe('testing mapStateToProps', () => {
  const state: any = { app: { scores: 'scores' } };
  it('test the mapping', () => {
    const expected = { scores: 'scores' };
    expect(mapStateToProps(state)).toEqual(expected);
  })
})

describe('testing mapDispatchToProps', () => {
  const dispatch = jest.fn();
  const mapper = mapDispatchToProps(dispatch);
  it('checks the keys', () => {
    expect(Object.keys(mapper)).toEqual(['resetScores']);
  })
  it('checks the dispatch functions', () => {
    mapper.resetScores();
    expect(dispatch).toHaveBeenCalledWith(setScores(null));
  })
})