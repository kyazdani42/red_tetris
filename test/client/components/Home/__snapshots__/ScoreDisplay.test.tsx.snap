// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the component smoke tests the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    resetScores={[Function]}
    scores={
      Array [
        Object {
          "bestScore": 10,
          "gamesPlay": 1,
          "multiPlayersWin": 1,
          "name": "testplayer",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <UNDEFINED>
        <UNDEFINED>
          Top 25 Scores
        </UNDEFINED>
        <UNDEFINED>
          <UNDEFINED
            style={
              Object {
                "marginBottom": "20px",
              }
            }
          >
            <span>
              <b>
                Player
              </b>
            </span>
            <span>
              <b>
                Top Score
              </b>
            </span>
            <span>
              <b>
                Games Played
              </b>
            </span>
            <span>
              <b>
                Multiplayer Game Wins
              </b>
            </span>
          </UNDEFINED>
          <UNDEFINED>
            <span>
              testplayer
            </span>
            <span>
              10
            </span>
            <span>
              1
            </span>
            <span>
              1
            </span>
          </UNDEFINED>
        </UNDEFINED>
      </UNDEFINED>,
      "className": "modal",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <UNDEFINED>
            Top 25 Scores
          </UNDEFINED>,
          <UNDEFINED>
            <UNDEFINED
              style={
                Object {
                  "marginBottom": "20px",
                }
              }
            >
              <span>
                <b>
                  Player
                </b>
              </span>
              <span>
                <b>
                  Top Score
                </b>
              </span>
              <span>
                <b>
                  Games Played
                </b>
              </span>
              <span>
                <b>
                  Multiplayer Game Wins
                </b>
              </span>
            </UNDEFINED>
            <UNDEFINED>
              <span>
                testplayer
              </span>
              <span>
                10
              </span>
              <span>
                1
              </span>
              <span>
                1
              </span>
            </UNDEFINED>
          </UNDEFINED>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Top 25 Scores",
          },
          "ref": null,
          "rendered": "Top 25 Scores",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-gqjmRU",
              "isStatic": true,
              "rules": Array [
                "
  color: #fff;
  text-align: center;
  margin-bottom: 30px;
",
              ],
            },
            "displayName": "styled.h1",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-gqjmRU",
            "target": "h1",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <UNDEFINED
                style={
                  Object {
                    "marginBottom": "20px",
                  }
                }
              >
                <span>
                  <b>
                    Player
                  </b>
                </span>
                <span>
                  <b>
                    Top Score
                  </b>
                </span>
                <span>
                  <b>
                    Games Played
                  </b>
                </span>
                <span>
                  <b>
                    Multiplayer Game Wins
                  </b>
                </span>
              </UNDEFINED>,
              Array [
                <UNDEFINED>
                  <span>
                    testplayer
                  </span>
                  <span>
                    10
                  </span>
                  <span>
                    1
                  </span>
                  <span>
                    1
                  </span>
                </UNDEFINED>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <span>
                    <b>
                      Player
                    </b>
                  </span>,
                  <span>
                    <b>
                      Top Score
                    </b>
                  </span>,
                  <span>
                    <b>
                      Games Played
                    </b>
                  </span>,
                  <span>
                    <b>
                      Multiplayer Game Wins
                    </b>
                  </span>,
                ],
                "style": Object {
                  "marginBottom": "20px",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b>
                      Player
                    </b>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Player",
                    },
                    "ref": null,
                    "rendered": "Player",
                    "type": "b",
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b>
                      Top Score
                    </b>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Top Score",
                    },
                    "ref": null,
                    "rendered": "Top Score",
                    "type": "b",
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b>
                      Games Played
                    </b>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Games Played",
                    },
                    "ref": null,
                    "rendered": "Games Played",
                    "type": "b",
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b>
                      Multiplayer Game Wins
                    </b>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Multiplayer Game Wins",
                    },
                    "ref": null,
                    "rendered": "Multiplayer Game Wins",
                    "type": "b",
                  },
                  "type": "span",
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-jTzLTM",
                  "isStatic": true,
                  "rules": Array [
                    "
  border-bottom: 1px solid rgba(255,255,255,0.2);
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  text-align: center;
  width: 95%;
  margin: auto;
  color: #fff;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-jTzLTM",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": "score-nb-0",
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <span>
                    testplayer
                  </span>,
                  <span>
                    10
                  </span>,
                  <span>
                    1
                  </span>,
                  <span>
                    1
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "testplayer",
                  },
                  "ref": null,
                  "rendered": "testplayer",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 10,
                  },
                  "ref": null,
                  "rendered": 10,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 1,
                  },
                  "ref": null,
                  "rendered": 1,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 1,
                  },
                  "ref": null,
                  "rendered": 1,
                  "type": "span",
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-jTzLTM",
                  "isStatic": true,
                  "rules": Array [
                    "
  border-bottom: 1px solid rgba(255,255,255,0.2);
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  text-align: center;
  width: 95%;
  margin: auto;
  color: #fff;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-jTzLTM",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-VigVT",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  flex-direction: column;
  overflow-y: scroll;
  height: fit-content;
  width: 100%;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-VigVT",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-gZMcBi",
          "isStatic": true,
          "rules": Array [
            "
  position: relative;
  margin: auto;
  top: 5%;
  height: 90%;
  width: 70%;
  border: 1px solid rgba(255, 255, 255, 0.4);
  background-color: #000;
  border-radius: 5px;
  box-shadow: 0 10px 10px rgba(255, 44, 55, 0.2), 0 5px 5px rgba(255, 55, 55, 0.2);
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-gZMcBi",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  height: 100vh;
  width: 100%;
  z-index: 1000;
  background-color: rgba(0, 0, 0, 0.7);
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <UNDEFINED>
          <UNDEFINED>
            Top 25 Scores
          </UNDEFINED>
          <UNDEFINED>
            <UNDEFINED
              style={
                Object {
                  "marginBottom": "20px",
                }
              }
            >
              <span>
                <b>
                  Player
                </b>
              </span>
              <span>
                <b>
                  Top Score
                </b>
              </span>
              <span>
                <b>
                  Games Played
                </b>
              </span>
              <span>
                <b>
                  Multiplayer Game Wins
                </b>
              </span>
            </UNDEFINED>
            <UNDEFINED>
              <span>
                testplayer
              </span>
              <span>
                10
              </span>
              <span>
                1
              </span>
              <span>
                1
              </span>
            </UNDEFINED>
          </UNDEFINED>
        </UNDEFINED>,
        "className": "modal",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <UNDEFINED>
              Top 25 Scores
            </UNDEFINED>,
            <UNDEFINED>
              <UNDEFINED
                style={
                  Object {
                    "marginBottom": "20px",
                  }
                }
              >
                <span>
                  <b>
                    Player
                  </b>
                </span>
                <span>
                  <b>
                    Top Score
                  </b>
                </span>
                <span>
                  <b>
                    Games Played
                  </b>
                </span>
                <span>
                  <b>
                    Multiplayer Game Wins
                  </b>
                </span>
              </UNDEFINED>
              <UNDEFINED>
                <span>
                  testplayer
                </span>
                <span>
                  10
                </span>
                <span>
                  1
                </span>
                <span>
                  1
                </span>
              </UNDEFINED>
            </UNDEFINED>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Top 25 Scores",
            },
            "ref": null,
            "rendered": "Top 25 Scores",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-gqjmRU",
                "isStatic": true,
                "rules": Array [
                  "
  color: #fff;
  text-align: center;
  margin-bottom: 30px;
",
                ],
              },
              "displayName": "styled.h1",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-gqjmRU",
              "target": "h1",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <UNDEFINED
                  style={
                    Object {
                      "marginBottom": "20px",
                    }
                  }
                >
                  <span>
                    <b>
                      Player
                    </b>
                  </span>
                  <span>
                    <b>
                      Top Score
                    </b>
                  </span>
                  <span>
                    <b>
                      Games Played
                    </b>
                  </span>
                  <span>
                    <b>
                      Multiplayer Game Wins
                    </b>
                  </span>
                </UNDEFINED>,
                Array [
                  <UNDEFINED>
                    <span>
                      testplayer
                    </span>
                    <span>
                      10
                    </span>
                    <span>
                      1
                    </span>
                    <span>
                      1
                    </span>
                  </UNDEFINED>,
                ],
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <span>
                      <b>
                        Player
                      </b>
                    </span>,
                    <span>
                      <b>
                        Top Score
                      </b>
                    </span>,
                    <span>
                      <b>
                        Games Played
                      </b>
                    </span>,
                    <span>
                      <b>
                        Multiplayer Game Wins
                      </b>
                    </span>,
                  ],
                  "style": Object {
                    "marginBottom": "20px",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <b>
                        Player
                      </b>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Player",
                      },
                      "ref": null,
                      "rendered": "Player",
                      "type": "b",
                    },
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <b>
                        Top Score
                      </b>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Top Score",
                      },
                      "ref": null,
                      "rendered": "Top Score",
                      "type": "b",
                    },
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <b>
                        Games Played
                      </b>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Games Played",
                      },
                      "ref": null,
                      "rendered": "Games Played",
                      "type": "b",
                    },
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <b>
                        Multiplayer Game Wins
                      </b>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Multiplayer Game Wins",
                      },
                      "ref": null,
                      "rendered": "Multiplayer Game Wins",
                      "type": "b",
                    },
                    "type": "span",
                  },
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-jTzLTM",
                    "isStatic": true,
                    "rules": Array [
                      "
  border-bottom: 1px solid rgba(255,255,255,0.2);
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  text-align: center;
  width: 95%;
  margin: auto;
  color: #fff;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-jTzLTM",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": "score-nb-0",
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <span>
                      testplayer
                    </span>,
                    <span>
                      10
                    </span>,
                    <span>
                      1
                    </span>,
                    <span>
                      1
                    </span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "testplayer",
                    },
                    "ref": null,
                    "rendered": "testplayer",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 10,
                    },
                    "ref": null,
                    "rendered": 10,
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 1,
                    },
                    "ref": null,
                    "rendered": 1,
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 1,
                    },
                    "ref": null,
                    "rendered": 1,
                    "type": "span",
                  },
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-jTzLTM",
                    "isStatic": true,
                    "rules": Array [
                      "
  border-bottom: 1px solid rgba(255,255,255,0.2);
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  text-align: center;
  width: 95%;
  margin: auto;
  color: #fff;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-jTzLTM",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-VigVT",
                "isStatic": true,
                "rules": Array [
                  "
  display: flex;
  flex-direction: column;
  overflow-y: scroll;
  height: fit-content;
  width: 100%;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-VigVT",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gZMcBi",
            "isStatic": true,
            "rules": Array [
              "
  position: relative;
  margin: auto;
  top: 5%;
  height: 90%;
  width: 70%;
  border: 1px solid rgba(255, 255, 255, 0.4);
  background-color: #000;
  border-radius: 5px;
  box-shadow: 0 10px 10px rgba(255, 44, 55, 0.2), 0 5px 5px rgba(255, 55, 55, 0.2);
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-gZMcBi",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  height: 100vh;
  width: 100%;
  z-index: 1000;
  background-color: rgba(0, 0, 0, 0.7);
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
